pipeline/alert.go

```diff
@@ -1254,3 +1258,76 @@ func (a *AlertNode) Talk() *TalkHandler {
 type TalkHandler struct {
 	*AlertNode
 }
+
+// Send the alert using SNMP traps.
+// To allow Kapacitor to post SNMP traps,
+//
+// Example:
+//    [snmp]
+//      enabled = true
+//      target-ip = "127.0.0.1"
+//      target-port = 9162
+//      community = "public"
+//      version = "2c"
+//      state-changes-only = false
+//
+// In order to not post a message every alert interval
+// use AlertNode.StateChangesOnly so that only events
+// where the alert changed state are posted to the channel.
+//
+// Example:
+//    stream
+//         |alert()
+//             .snmp()
+//             .trap('1.3.6.1.2.1.1.7', 'i', {{ Index field value }})
+//
+// Send alerts to `target-ip:target-port` on OID '1.3.6.1.2.1.1.7'
+//
+// tick:property
+func (a *AlertNode) Snmp(trap ...[]interface{}) *SnmpHandler {
+	snmp := &SnmpHandler{
+		AlertNode: a,
+		TrapList:  trap,
+	}
+	a.SnmpHandlers = append(a.SnmpHandlers, snmp)
+	return snmp
+}
+
+// Email AlertHandler
+// tick:embedded:AlertNode.Email
+type SnmpHandler struct {
+	*AlertNode
+
+	// List of email recipients.
+	// tick:ignore
+	TrapList [][]interface{} `tick:"Trap"`
```

Comment:
What information does a trap need?

Based on the configuration and the docs, it looks like traps have a these fields:

* Target IP
* Target Port
* OID
* Version
* Community

Anything else?

Maybe changing the syntax up a bit would make it more clear:

```
|alert()
    .snmpTrap()
       .ip('x.x.x.x')
       .port(9863)
       .oid('1.3.6.1.2.1.1.7')
       .version('v3')
       .community('asdf')
    .snmpTrap()
       .ip('x.x.x.x')
       .port(2656)
       .oid('1.3.6.1.2.1.1.7')
       .version('v2')
    .snmpTrap()
       // defaults to config values if not specified
       .oid('1.3.6.1.2.1.1.7')
```